iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlPreprocessor.java

```diff
@@ -53,4 +54,12 @@ private static String removeInlineCommentAfterDocumentBreak(String line) {
     }
     return line;
   }
+
+  /**
+   * Surround Maven variable substitution in quotes instead of '@' to avoid parsing issues.
+   * We don't resolve these variables, so we shouldn't lose information by doing this.
+   */
+  private static String transformMavenSubstitutions(String line) {
+    return MAVEN_VARIABLE_AT_TOKEN_START.matcher(line).replaceAll("$1'${name}'$3");
```

Comment:
I was thinking about either substituting `@...@` with `${,,,}`, but that would require tracking text range adjustments, or tracking the information about the substituted symbols and replacing them back in the converter. However, I don't see an immediate value from it, because it's not likely that maven variable name will clash with a sensitive property value, where we would raise an issue.

Reply:
Yes, it's probably better not to mess with offset. This is generally not so fun to handle, as shown with our docker preprocessor.
Seems good to me to keep it as it is for now, until we get additional values from implementing a more complete/complex solution.